R/time_converters.R

```diff
@@ -0,0 +1,127 @@
+#' convert an archive from daily data to weekly data, summing where appropriate
+#' @details
+#' this function is slow, so make sure you are calling it correctly, and
+#'   consider testing it on a small portion of your archive first
+#' @param day_of_week integer, day of the week, starting from Sunday, select the
+#'   date to represent the week in the time_value column, based on it's
+#'   corresponding day of the week. The default value represents the week using
+#'   Wednesday.
+#' @param day_of_week_end integer, day of the week starting on Sunday.
+#'   Represents the last day, so the week consists of data summed to this day.
+#'   The default value `6` means that the week is summed from Sunday through
+#'   Saturday.
+daily_to_weekly <- function(epi_arch,
+                            agg_columns,
+                            agg_method = c("sum", "mean"),
+                            day_of_week = 4L,
+                            day_of_week_end = 6L) {
+  agg_method <- arg_match(agg_method)
+  if (agg_method == "total") {
+    agg_fun <- epi_slide_sum
+  } else if (agg_method == "mean") {
+    agg_fun <- epi_slide_mean
+  }
+  keys <- grep("time_value", key_colnames(epi_arch), invert = TRUE, value = TRUE)
+  too_many_tibbles <- epix_slide(
+    epi_arch,
+    before = 99999999L,
+    ref_time_values = ref_time_values,
+    function(x, group, ref_time) {
+      x %>%
+        group_by(across(all_of(keys))) %>%
```

Comment:
after new updates you can just do and get rid of the keys assignment above
```suggestion
        group_epi_df(exclude = "time_value") %>%
```

Reply:
glad to hear it! I was getting very tired of recreating `keys` all over the place